### PR TITLE
wit: add more keywords to the escape list

### PR DESCRIPTION
**Pull request by Mossaka**
while i am trying to add more runtime tests from the wit-bindgen repo to here, I discovered that there are a lot of words such as `bool` are not being covered. I also found the package names are not being escaped.

This commit fixes both issues

I am intended to fix all the go test bugs in this PR and so far there are these:

- [ ] https://github.com/bytecodealliance/wasm-tools-go/issues/160
- [ ] https://github.com/bytecodealliance/wasm-tools-go/issues/161

### CODE REVIEW

File: wit/ident.go [hunk 1]
```diff
@@ -38,6 +38,7 @@ func ParseIdent(s string) (Ident, error) {
 	name, ver, hasVer := strings.Cut(s, "@")
 	base, ext, hasExt := strings.Cut(name, "/")
 	id.Namespace, id.Package, _ = strings.Cut(base, ":")
```
Comment (ydnar) [inline wit/ident.go:40]: I'd put these into temp vars stage then mutate id twice.

File: wit/wit.go [hunk 2]
```diff
@@ -946,16 +972,19 @@ func (f *Function) WIT(ctx Node, name string) string {
 	case worldExport:
 		b.WriteString("export ")
 	}
-	b.WriteString(escape(name))
 	var isConstructor, isMethod bool
 	switch f.Kind.(type) {
 	case *Constructor:
+		// constructor is a keyword in WIT, but should not be escaped as a function name
```
Comment (ydnar) [inline wit/wit.go:978]: 👍

**Review by ydnar — CHANGES_REQUESTED** [review]
Thanks! This is great. 

Can you split into two PRs: one that makes the changes (with small test changes as necessary) and a second PR that introduces the large number of new test files?

**Comment by Mossaka** [comment]
> Can you split into two PRs: one that makes the changes (with small test changes as necessary) and a second PR that introduces the large number of new test files?

This PR only adds 4 test files. Are you suggesting to trim down the test files further?

**Comment by ydnar** [comment]
@Mossaka your PR uncovered some interesting bugs, thanks!

**Review by ydnar — CHANGES_REQUESTED** [review]
Need to update CHANGELOG, but that’s it!

**Comment by Mossaka** [comment]
Done

**Review by ydnar — APPROVED** [review]

**Pull request merged**